packages/react/src/auto/shadcn/inputs/ShadcnAutoTextInput.tsx

```diff
@@ -0,0 +1,43 @@
+import React from "react";
+import type { InputHTMLAttributes } from "react";
+import type { Control } from "../../../useActionForm.js";
+import { getPropsWithoutRef } from "../../../utils.js";
+import { autoInput } from "../../AutoInput.js";
+import { useStringInputController } from "../../hooks/useStringInputController.js";
+import type { ShadcnElements } from "../elements.js";
+
+// should we move this to a shared location?
+import { cn } from "../../../../spec/auto/shadcn-defaults/utils.js";
```

Comment:
I think the move would be to expect this to be passed in as an element that you can then get from the context, as users are likely gonna have their own version that they use in their shadcn components, and we wnat to use the same one. generally speaking, any import from the `spec` dir in the `src` dir is busted

Reply:
@airhorns This import was to merge the styles that came with the users component and add an error state if the input had an error.

Reply:
Right, I understant we need the utility, but I am saying we can't import it ourselves from the version in `spec`. I think this utility needs to be provided to us as one of the elements, and then referred to from the elements at runtime. The user will need their own version of this utility in their app, and will use it in all their shadcn components, we should re-use theirs, not import from our own. In general, any code in `src` importing from `spec` is an antipattern as well, if we intend to ship the code at runtime it needs to be in `src`. But, that's why I put the default components in `spec` -- we don't intend to ship any of that at runtime, its just a copy of shadcn we use for testing.